src/tsf/ConsoleTSF.cpp

```diff
@@ -268,24 +268,24 @@ STDMETHODIMP CConsoleTSF::OnCleanupContext(TfEditCookie ecWrite, ITfContext* pic
     //
     // Remove GUID_PROP_COMPOSING
     //
-    CComPtr<ITfProperty> prop;
+    wil::com_ptr_nothrow<ITfProperty> prop;
     if (SUCCEEDED(pic->GetProperty(GUID_PROP_COMPOSING, &prop)))
     {
-        CComPtr<IEnumTfRanges> enumranges;
-        CComPtr<ITfRange> rangeFull;
-        if (SUCCEEDED(prop->EnumRanges(ecWrite, &enumranges, rangeFull)))
+        wil::com_ptr_nothrow<IEnumTfRanges> enumranges;
+        wil::com_ptr_nothrow<ITfRange> rangeFull;
+        if (SUCCEEDED(prop->EnumRanges(ecWrite, &enumranges, rangeFull.get())))
         {
-            CComPtr<ITfRange> rangeTmp;
+            wil::com_ptr_nothrow<ITfRange> rangeTmp;
             while (enumranges->Next(1, &rangeTmp, NULL) == S_OK)
             {
                 VARIANT var;
                 VariantInit(&var);
-                prop->GetValue(ecWrite, rangeTmp, &var);
+                prop->GetValue(ecWrite, rangeTmp.get(), &var);
                 if ((var.vt == VT_I4) && (var.lVal != 0))
                 {
-                    prop->Clear(ecWrite, rangeTmp);
+                    prop->Clear(ecWrite, rangeTmp.get());
                 }
-                rangeTmp.Release();
+                rangeTmp.reset();

```

Comment:
`reset` isn't needed in situations like this one (there are a few in the TSF code) - `wil::com_ptr::operator&` already does a `reset`.